kafka-admin/src/main/java/io/strimzi/admin/kafka/admin/handlers/TopicListHandler.java

```diff
@@ -1,41 +1,57 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
 package io.strimzi.admin.kafka.admin.handlers;
 
-import graphql.schema.DataFetchingEnvironment;
-import io.strimzi.admin.kafka.admin.model.Topic;
+import io.strimzi.admin.kafka.admin.AdminClientProvider;
+import io.strimzi.admin.kafka.admin.model.Types;
 import io.vertx.core.Promise;
-import java.util.HashMap;
+import io.vertx.ext.web.handler.graphql.VertxDataFetcher;
+import io.vertx.ext.web.impl.RoutingContextWrapper;
+
+import java.util.ArrayList;
 import java.util.List;
-import java.util.Map;
+import java.util.Set;
 import java.util.function.Predicate;
-import java.util.stream.Collectors;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class TopicListHandler {
 
-    public static final Map<String, Topic> SAMPLE_TOPICS = new HashMap<>();
-    static {
-        SAMPLE_TOPICS.put("topic1", Topic.create("topic1", false, 3, 3));
-        SAMPLE_TOPICS.put("topic2", Topic.create("topic2", false, 1, 1));
-        SAMPLE_TOPICS.put("topic2-a", Topic.create("topic2-a", false, 1, 1));
-        SAMPLE_TOPICS.put("topic2-b", Topic.create("topic2-b", false, 1, 1));
-        SAMPLE_TOPICS.put("topic2-c", Topic.create("topic2-c", false, 1, 1));
-        SAMPLE_TOPICS.put("topic3", Topic.create("topic3", false, 6, 1));
-    }
+    public static VertxDataFetcher topicListFetch(AdminClientProvider acp) {
+        VertxDataFetcher<Types.TopicList> dataFetcher = new VertxDataFetcher<>((env, prom) -> {
+            RoutingContextWrapper routingContext = env.getContext();
+            routingContext.request().headers().get("token");
```

Comment:
why this "token" handling just here and not in all the other handlers? I am not saying we should add to everyone but we should first test how/if it works. You are also just getting without save/use it?

Reply:
Yes. I know we want to use them for authn... so it is just prepared to be used. I will remove it for now.